packages/i18nify-js/src/modules/phoneNumber/getMaskedPhoneNumber.ts

```diff
@@ -0,0 +1,41 @@
+import getDialCodeByCountryCode from './getDialCodeByCountryCode';
+import { withErrorBoundary } from '../../common/errorBoundary';
+import PHONE_FORMATTER_MAPPER from './data/phoneFormatterMapper.json';
+import { CountryCodeType } from '../types';
+
+/**
+ * Returns a masked phone number based on the country code.
+ *
+ * @param countryCode The ISO 3166-1 alpha-2 country code.
+ * @param withDialCode A boolean indicating whether to include the country's dial code in the result. It has a default value of "true"
+ * @returns The masked phone number as a string.
+ */
+const getMaskedPhoneNumber = (
```

Comment:
What is the use case of this right now ? 
Let's think more from open source perspective. Can we extend this utility in future to also allow partial masking of phone numbers ? something like `70424 XXXXX` which is a common use case. 

In this we need to design the schema accordingly so it's backward compatible.

Reply:
Discussed over call.